src/mrpro/data/_kdata/KData.py

```diff
@@ -234,3 +234,17 @@ def sort_and_reshape_tensor_fields(input_tensor: torch.Tensor):
             ) from None
 
         return cls(kheader, kdata, ktrajectory_final)
+
+    def __repr__(self):
+        """Representation method for KData class."""
+        traj = str(self.traj.as_tensor().shape).replace('torch.Size', '')
+        device_traj = 'cuda' if self.traj.device == torch.device('cuda') else 'cpu'
```

Comment:
Maybe we could replace this by something like
```
try:
    device = self.device
except InconsistentDeviceError:
    device= 'mixed'
```
The main advantage is that it also works if there are other tensors apart from traj and data. We could then use the same for QData+IData (or adding it to Data), KNoise and DcfData.

Reply:
I think this answers your question @Stef-Martin, correct?

Reply:
you can also just do str(device) to the device string:
for CPU it will be CPU
for GPU it will be cuda, followed by the GPU number (which is also useful)

Reply:
yeah, i was just confused by InconsistentDeviceError - but turns out I overthought this a bit and this just gives a RuntimeError, which is what i included as the check now :D